src/main/java/com/amazonaws/sagemaker/controller/ServingController.java

```diff
@@ -148,6 +150,31 @@ public ResponseEntity returnBatchExecutionParameter() throws JsonProcessingExcep
         }
     }
 
+    /**
+     * Implements the invocations POST API for application/jsonlines input
+     *
+     * @param jsonLines, lines of json values
+     * @param accept, indicates the content types that the http method is able to understand
+     * @return ResponseEntity with body as the expected payload JSON & proper statuscode based on the input
+     */
+    @RequestMapping(path = "/invocations", method = POST, consumes = AdditionalMediaType.APPLICATION_JSONLINES_VALUE)
+    public ResponseEntity<String> transformRequestJsonLines(@RequestBody final byte[] jsonLines,
+                                                            @RequestHeader(value = HttpHeaders.ACCEPT, required = false) final String accept) {
+        final byte[] emptyInput = new byte[0];
```

Comment:
You can just do jsonLines.length == 0 right? Doesnt that work?

Reply:
And don't we need to check for null as well? I'd recommend explicitly handling the null input case and returning a bad request instead of just letting it flow and raising an exception.